05.wc/wc.rb

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+DEFAULT_WIDTH = 8
+
+def main
+  options = ARGV.getopts('lwc')
+  ARGV.none? ? print_from_stdin(options) : print_from_args(ARGV, options)
+end
+
+def print_from_stdin(options)
+  input_data_lines = $stdin.readlines
+
+  lines_num = count_lines(input_data_lines)
+  words_num = count_words(input_data_lines)
+  file_size = count_bytesize(input_data_lines)
+  total_nums = { 'total_lines' => lines_num, 'total_words' => words_num, 'total_size' => file_size }
+
+  print_total(total_nums, options)
+  print "\n"
+end
+
+def print_from_args(files, options)
+  files_info = format_files_info(files)
+
+  if files.size > 1
+    print_info(files_info[:lines_words_sizes], files_info[:total_nums], options)
+    print_total(files_info[:total_nums], options)
+    print " total\n"
+  else
+    print_info(files_info[:lines_words_sizes], files_info[:total_nums], options)
+  end
+end
+
+def format_files_info(files)
+  all_files_lines = files.map { |file| File.open(file, &:readlines) }
+
+  lines_nums = all_files_lines.map { |file_lines| count_lines(file_lines) }
+  total_lines_nums = lines_nums.sum
+
+  words_nums = all_files_lines.map { |file_lines| count_words(file_lines) }
+  total_words_nums = words_nums.sum
+
+  file_sizes = all_files_lines.map { |file_lines| count_bytesize(file_lines) }
+  total_size = file_sizes.sum
+
+  lines_words_sizes = [lines_nums, words_nums, file_sizes, files].transpose
+  lines_words_sizes_keys = %w[lines_num words_num file_size file_name]
+  formatted_3_elemetnts_info =
+    lines_words_sizes.map { |file_info| [lines_words_sizes_keys, file_info].transpose.to_h }
```

Comment:
```suggestion
    lines_words_sizes.map do |file_info|
      { lines_num: lines_nums, ... }
    end
```

というようにハッシュリテラルを使った方がシンプルで読みやすそうです。

Ruby 3.1以降なら、

```suggestion
    lines_words_sizes.map do |file_info|
      { lines_num:, ... }
    end
```

のようにして変数名とキー名をまとめることもできます。

[Ruby 3\.1: ES6風のハッシュリテラル省略記法が追加（翻訳）｜TechRacho by BPS株式会社](https://techracho.bpsinc.jp/hachi8833/2021_11_24/113383)

Reply:
こちらご教示ありがとうございます🙇‍♂️

```ruby
    lines_words_sizes.map do |file_info|
      { lines_num: lines_nums, ... }
    end
```
提案いただいたこの部分ですが、ブロックに渡している`file_info`は`transpose`した後の配列なので、提案いただいたような形でハッシュリテラルにうまく渡すことができません🙇‍♂️

https://github.com/junohm410/ruby-practices/blob/714b462bd5f781def723017c58a06a4c4f04e10c/05.wc/wc.rb#L49-L53

（変数名は変わっていますが、）このようにブロックに渡した配列の個々の要素を取得するようにすれば、ハッシュリテラルにうまく値をパスできるのですが、
これはそもそも前回指摘いただいた「配列の順番に意味を持たせてはならない」に違反しており、よくないのかなと思っております🤧

伊藤さんからサジェストしていただいている内容を私がうまく汲み取れていないのかも知れないと思いますが、現時点で私の中で手詰まりなので、上記内容で一旦修正して提出しております。
取り違えている点や解決のヒントをもしご教示いただけるのであればありがたいです🙇‍♂️